quinn-proto/src/connection/mod.rs

```diff
@@ -362,6 +435,85 @@ impl Connection {
         this
     }
 
+    /// Set up qlog for this connection.
+    #[cfg(feature = "qlog")]
+    pub fn set_qlog(
+        &mut self,
+        writer: Box<dyn std::io::Write + Send + Sync>,
+        title: String,
+        description: String,
+    ) {
+        let vp = if self.side.is_server() {
+            qlog::VantagePointType::Server
+        } else {
+            qlog::VantagePointType::Client
+        };
+
+        let level = qlog::events::EventImportance::Core;
+
+        let trace = qlog::TraceSeq::new(
+            qlog::VantagePoint {
+                name: None,
+                ty: vp,
+                flow: None,
+            },
+            Some(title.to_string()),
+            Some(description.to_string()),
+            Some(qlog::Configuration {
+                time_offset: Some(0.0),
+                original_uris: None,
+            }),
+            None,
+        );
+
+        let mut streamer = qlog::streamer::QlogStreamer::new(
+            qlog::QLOG_VERSION.to_string(),
+            Some(title),
+            Some(description),
+            None,
+            std::time::Instant::now(),
```

Comment:
To aid deterministic testing, we prefer not to call `Instant::now` in `quinn-proto`, instead letting the caller pass an `Instant` in. Should we preserve that pattern here? Might not matter since this is just a diagnostic, but deterministic qlog traces could conceivably be a useful tool.